storm-client/src/jvm/org/apache/storm/executor/spout/SpoutExecutor.java

```diff
@@ -105,19 +107,21 @@ public void expire(Long key, TupleInfo tupleInfo) {
             }
         });
 
-        this.spoutThrottlingMetrics.registerAll(topoConf, idToTask.values().iterator().next().getUserContext());
+        this.spoutThrottlingMetrics.registerAll(topoConf, idToTask.get(taskIds.get(0)).getUserContext());
         this.outputCollectors = new ArrayList<>();
-        for (Map.Entry<Integer, Task> entry : idToTask.entrySet()) {
-            Task taskData = entry.getValue();
+        for (int i=0; i<idToTask.size(); ++i) {
```

Comment:
- Lookups occur in derived classes
- idToTask.entrySet() can cause object allocation.